main.tf

```
@@ -83,15 +84,13 @@ data "aws_iam_policy_document" "monitoring_rds_assume_role" {
 }
 
 resource "aws_iam_role" "rds_enhanced_monitoring" {
-  count = "${var.monitoring_interval > 0 ? 1 : 0}"
-
+  count              = "${var.create_resources ? var.monitoring_interval > 0 ? 1 : 0 : 0}"
```

Less comparisons is nicer, I believe:
```suggestion
  count              = "${var.create_resources && var.monitoring_interval > 0 ? 1 : 0}"
```